src/util/psbt/responsibilities.rs

```diff
@@ -0,0 +1,133 @@
+use hashes::core::fmt::Debug;
+use secp256k1::{Message, Signature};
+use util::bip32::{DerivationPath, ExtendedPrivKey, Fingerprint};
+use util::psbt::PartiallySignedTransaction;
+use {Address, Amount};
+use {Script, TxOut};
+
+/// This trait corresponds to the Creator and Updater responsibility described in
+/// [BIP-174](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki#creator). As noted
+/// there the both functionalities are most often combined as the creator most likely has more meta
+/// information about the transaction they are creating and is thus able to put it into the PSBT
+/// (which otherwise would be a separate update step).
+pub trait PsbtWallet {
+    /// Wallet backend specific errors like database connection errors that aren't captured in
+    /// `PsbtCreationError`.
+    type Error: Debug;
+
+    /// Create a transaction that pays a set of outputs.
+    fn create_transaction(
+        &mut self,
+        outputs: &[TxOut],
+    ) -> Result<PartiallySignedTransaction, PsbtCreationError<Self::Error>>;
```

Comment:
Is this method supposed to create a transaction without inputs or is the implementor responsible for adding inputs to fulfill the tx?

Reply:
Yes, it's the implementor's responsibility to select UTXOs, that's an integral part of every wallet imo: tracking and selecting UTXOs.